components/script/dom/abortsignal.rs

```diff
@@ -37,7 +38,7 @@ pub(crate) enum AbortAlgorithm {
     /// <https://streams.spec.whatwg.org/#readable-stream-pipe-to>
     StreamPiping(PipeTo),
     /// <https://fetch.spec.whatwg.org/#dom-global-fetch>
-    Fetch,
+    Fetch(FetchContext),
```

Comment:
This needs a struct, not a trait. So I think we need the full `FetchContext` here? I am confused why this works for `PipeTo`, but wouldn't work for this struct.

Reply:
I think this problem goes away if you use the `Arc<Mutex<FetchContext>>` that's required elsewhere.